src/main/java/com/example/springbootapp/exception/EmployeeExceptionHandler.java

```diff
@@ -33,6 +38,44 @@ public ResponseError employeeNotSaved(EmployeeNotSavedException exception) {
         return new ResponseError(HttpStatus.BAD_REQUEST, exception.getMessage());
     }
 
+    @ResponseBody
+    @ExceptionHandler(ConstraintViolationException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ResponseError employeeConstraintViolation(
+            ConstraintViolationException exception
+    ) {
+        final List<Violation> violations = extractConstraintViolations(exception);
+        return new ResponseError(HttpStatus.BAD_REQUEST, violations);
+    }
+
+    private List<Violation> extractConstraintViolations(ConstraintViolationException exception) {
+        LOGGER.error(exception.getMessage(), exception);
```

Comment:
Please, move logging into the main method(employeeConstraintViolation). The same for the next method

Reply:
I moved it.